tests/granulate_utils/test_cgroups.py

```diff
@@ -1,28 +1,231 @@
-from typing import Callable
+#
+# Copyright (c) Granulate. All rights reserved.
+# Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
+#
+
+from pathlib import Path
 from unittest.mock import patch
 
-from granulate_utils.linux.cgroups.cpu_cgroup import CpuCgroup
-
-
-@patch("granulate_utils.linux.cgroups.cpu_cgroup.BaseCgroup._verify_preconditions", return_value=None)
-def test_get_cpu_limit_cores(base_group_mock: Callable) -> None:
-    cpucgroup = CpuCgroup()
-    with patch.object(
-        cpucgroup,
-        "read_from_control_file",
-        {
-            CpuCgroup.cfs_period_us: "100",
-            CpuCgroup.cfs_quota_us: "50",
-        }.__getitem__,
-    ):
-        assert cpucgroup.get_cpu_limit_cores() == 0.5
-
-    with patch.object(
-        cpucgroup,
-        "read_from_control_file",
-        {
-            CpuCgroup.cfs_period_us: "100",
-            CpuCgroup.cfs_quota_us: "-1",
-        }.__getitem__,
-    ):
-        assert cpucgroup.get_cpu_limit_cores() == -1.0
+import pytest
+from pytest import TempPathFactory
+
+from granulate_utils.linux.cgroups.cgroup import CgroupCoreV1, CgroupCoreV2, get_cgroup_for_process
+from granulate_utils.linux.cgroups.cpu_controller import CpuController
+from granulate_utils.linux.cgroups.cpuacct_controller import CpuAcctController
+from granulate_utils.linux.cgroups.memory_controller import MemoryController
+
+
+# Cgroup
+def test_cgroup_sanity(tmp_path_factory: TempPathFactory):
+    tmp_dir = tmp_path_factory.mktemp("base_controller")
+    cgroup_dir = Path(tmp_dir)
+    cpu_procs = cgroup_dir / "cgroup.procs"
+
+    cpu_procs.write_text("1 2 3")
+
+    cgroup_v1 = CgroupCoreV1(cgroup_dir)
+
+    assert cgroup_v1.get_pids_in_cgroup() == set([1, 2, 3])
+    cgroup_v1.assign_process_to_cgroup(4)
+    assert cgroup_v1.get_pids_in_cgroup() == set([4])  # write_text overwrites by default.
+
+    sub_cgroup_dir = cgroup_dir / "sub_cgroup"
+    assert not sub_cgroup_dir.exists()
+    sub_cgroup = cgroup_v1.get_cgroup_in_hierarchy("dummy", "sub_cgroup")
```

Comment:
```suggestion
    sub_cgroup = cgroup_v1.get_cgroup_in_hierarchy("dummy_controller", "sub_cgroup")
```

Or better, add a global const `DUMMY_CONTROLLER`. Then, reading the test code makes it perfectly clear that the value is the same dummy used around, and that you don't care about it. Without it, I now tried to look around if the `dummy` value is used anywhere, before I recalled it's the `controller` name and it's not relevant for v1.